internal/doc/header.md

```diff
@@ -5,6 +5,7 @@ nav_order: {{ .NavOrder }}
 ---
 
 # Configuration Specification v{{.Version}}
+<div id="spec-docs"></div>
```

Comment:
Legibility nit: can we move this after the `PreRelease` check, and ensure it has blank lines around it in the templates?

Reply:
Sure, will move it. I originally put it next to the heading because that way it was easier to keep it in sync with butane. Also having it there ensures that it's not handled as a lone, left-over fragment of something but part of the document structure "related to the heading".